samples/nextjs/src/lib/layout-personalization-service.ts

```diff
@@ -0,0 +1,84 @@
+import config from 'temp/config';
+import {
+  LayoutPersonalizationService,
+  RestPersonalizationDecisionsService,
+  GraphQLLayoutFragmentService,
+  LayoutServiceData,
+  isServer,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import { trackingService } from 'lib/tracking-service';
+import { NextRouter } from 'next/router';
+import { areQueryParamsReady } from './util';
+
+export const layoutPersonalizationService = new LayoutPersonalizationService(
+  new RestPersonalizationDecisionsService({
+    serviceUrl: config.personalizationDecisionsEndpoint,
+    apiKey: config.sitecoreServicesApiKey,
+    siteName: config.jssAppName,
+    tracking: config.trackingEnabled.toLocaleLowerCase() === 'true',
+    currentPageParamsToExtract: ['sc_camp', 'sc_trk'],
+  }),
+  new GraphQLLayoutFragmentService({
+    endpoint: config.graphQLEndpoint,
+    apiKey: config.sitecoreApiKey,
+    siteName: config.jssAppName,
+  })
+);
+
+export const loadPersonalization = async (
+  pageProps: { layoutData: LayoutServiceData; isPreview: boolean; tracked?: boolean },
+  router: NextRouter
+): Promise<void> => {
```

Comment:
The return type is labeled as a Promise type, but this is not always returning a Promise

Reply:
Async functions always return a promise. If the return value of an async function is not explicitly a promise, it will be implicitly wrapped in a promise.